Feature coordinates now saved to an array first, then saved to file.

msekatchev opened the pull request:
-	Changed annotate_img(): For each PMT recorded, the program now saves the coordinates to an array and writes the coordinates for the set of features of a PMT after recording for that PMT is finished.
-	This allows the user to increment/decrement and overwrite feature IDs.
-	Use b (back) and n (next) to increment/decrement the feature ID.
-	The program no longer manually increments the feature ID.
closes #9 "Implement incrementing/decrementing feature number manually"

msekatchev committed:
Feature coordinates now saved to an array first, then saved to file.

-	Changed annotate_img(): For each PMT recorded, the program now saves the coordinates to an array and writes the coordinates for the set of features of a PMT after recording for that PMT is finished.
-	This allows the user to increment/decrement and overwrite feature IDs.
-	Use b (back) and n (next) to increment/decrement the feature ID.
-	The program no longer manually increments the feature ID.
closes #9, "Implement incrementing/decrementing feature number manually"

msekatchev committed:
Feature coordinates now saved to an array first, then saved to file.

-	Changed annotate_img(): For each PMT recorded, the program now saves the coordinates to an array and writes the coordinates for the set of features of a PMT after recording for that PMT is finished.
-	This allows the user to increment/decrement and overwrite feature IDs.
-	Use b (back) and n (next) to increment/decrement the feature ID.
-	The program no longer manually increments the feature ID.
closes #9 "Implement incrementing/decrementing feature number manually"

msekatchev committed:
Feature coordinates now saved to an array first, then saved to file.

-	Changed annotate_img(): For each PMT recorded, the program now saves the coordinates to an array and writes the coordinates for the set of features of a PMT after recording for that PMT is finished.
-	This allows the user to increment/decrement and overwrite feature IDs.
-	Use b (back) and n (next) to increment/decrement the feature ID.
-	The program no longer manually increments the feature ID.
closes #9 "Implement incrementing/decrementing feature number manually"

msekatchev committed:
Merge branch 'master' into featureid_increment_decrement

pdeperio commented on README.txt:
```diff
@@ -9,22 +9,22 @@ def draw_circle(event,x,y,flags,param):
-        elif event == cv2.EVENT_LBUTTONUP:
-            drawing = False
 
+        #if event == cv2.EVENT_LBUTTONDOWN:
```
Wait, why does the `README.txt` have code in it now...?

(Eventually write a nice Markdown file, will make a new issue.)

msekatchev replied:
Will be solved in separate issue #19

pdeperio commented on annotate.py:
```diff
@@ -85,34 +87,70 @@ def annotate_img(img_path, initials, size1=1, size2=1) :
                 print("Recording ", writepmtID+"-"+writestartingVal, ix, "x", iy,"y\n")
-   
-                file.write("%s\t%s-%s\t%d\t%d\t%s\n" %(filename,writepmtID,writestartingVal, ix, iy,initials))
-                startingVal = startingVal+1   
```
I would keep the automatic incrementing after recording, otherwise need an extra keystroke for each feature (`r` + `n`).

msekatchev replied:
readded automatic incrementing 👍

pdeperio commented on annotate.py:
```diff
@@ -128,20 +166,26 @@ def annotate_img(img_path, initials, size1=1, size2=1) :
                     print("Recording ", writepmtID+"-"+writestartingVal, ix, "x", iy,"y\n")
                     print("Record another selected point by pressing r.\nPress x to skip recording a number.\nPress f to finish recording features for this PMT.")
```
Update screen instructions

msekatchev replied:
updated with increment and decrement keys.

msekatchev committed:
Feature coordinates now saved to an array first, then saved to file.

-	Changed annotate_img(): For each PMT recorded, the program now saves the coordinates to an array and writes the coordinates for the set of features of a PMT after recording for that PMT is finished.
-	This allows the user to increment/decrement and overwrite feature IDs.
-	Use b (back) and n (next) to increment/decrement the feature ID.
-	The program still automatically increments the feature ID.
closes #9 "Implement incrementing/decrementing feature number manually"


Changes to new featureid_increment_decrement commit:
readded automatic increment.
updated printed user instructions with keys for incrementing and decrementing.

msekatchev reviewed (commented):
Requests reviewed and implemented in new PR version.
